features/270-next-skycli/api.md

```diff
@@ -113,27 +113,144 @@ access skycli instead of using user account.
 
 ## Config files structure
 
-### ~/.skycli/skyclirc
+### ${XDG_CONFIG_HOME:-$HOME/.config}/skycli/config
 
 - cluster controller endpoint
 - user credentials 
 
-### ./.skyclirc
-- current app
-
-## Discussion
+### ./skygear.yaml
 
-- Current `skycli app view-config` and `skycli app update-config` give the
-greatest flexible to user to update the config. But user need to understand the
-tenant config structure. I think this case is similar to v1 cms config file.
-- Current tenant config is not a simple key value mapping, so we didn't have a
-single command to update particular value. User need to export the current config
-by `skycli app view-config`, update, and upload by `skycli app update-config`.
-- `skycli app view-config` will mask the auth db url, it is also not editable by
-`skycli app update-config`.
-- I think we should not show the secret value in the list comment. But user may
-want to obtain the db credentials after enabling service?
+- config version (If we have new config version in future, skycli will read the
+  version and convert the previous config to the current version.)
+- current app
+- cloud functions configuration
 
-## Questions
+#### Example
+```
+config_version: 1
+app: myapp
+cf:
+  function1:
+    type: http-handler
+    path: /hello-world
+    env: node
+    src: js/hellow-world
+    secrets:
+      - DATABASE_URL
+    permission:
+      - name: key_required
+      - name: user_required
+static:
+  ...
+```
 
-- How to support email template?
+## Use cases
+
+- User creates app and uses skycli for app configuration without CF
+
+    ```sh
+    # Create app
+    $ skycli app create
+
+    ? What is your app name? myapp
+    Creating app...
+    Your API endpoint: https://myapp.api.skygear.io
+    Your Client API Key: DSOJSDOSDKOJASNLSLC
+    Your Master API Key: FJOADJOFAJOFJOASDJK
+    Created app successfully!
+
+    To setup app project directory, run:
+        skycli app scaffold
```

Comment:
Ask user if they want to run app scaffold here? I guess 90% of the cases will require that? So we run `skycli app scaffold --app=[myapp]` in the current folder.